gcloud/monitoring/client.py

```diff
@@ -282,3 +283,89 @@ def list_resource_descriptors(self, filter_string=None):
             https://cloud.google.com/monitoring/api/v3/filters
         """
         return ResourceDescriptor._list(self, filter_string)
+
+    def group(self, group_id=None, name=None, display_name='', parent_name='',
+              filter_string='', is_cluster=False):
+        """Factory constructor for group object.
+
+        .. note::
+          This will not make an HTTP request; it simply instantiates
+          a group object owned by this client.
+
+        :type group_id: string
+        :param group_id:
+              The ID of the group. This is useful for reading or modifying an
+              existing group.
+
+        :type name: string or None
+        :param name:
+              The fully qualified name of the group.
+
+        :type display_name: string
+        :param display_name:
+            A user-assigned name for this group, used only for display
+            purposes.
+
+        :type parent_name: string
+        :param parent_name:
+            The fully qualified name of the group's parent, if it has one.
+
+        :type filter_string: string
+        :param filter_string:
+            The filter string used to determine which monitored resources
+            belong to this group.
+
+        :type is_cluster: boolean
+        :param is_cluster:
+            If true, the members of this group are considered to be a cluster.
+            The system can perform additional analysis on groups that are
+            clusters.
+
+        :rtype: :class:`Group`
+        :returns: The group created with the passed-in arguments.
+
+        :raises:
+            :exc:`ValueError` if both ``group_id`` and ``name`` are specified.
+        """
+        return Group(
+            self,
+            group_id=group_id,
+            name=name,
+            display_name=display_name,
+            parent_name=parent_name,
+            filter_string=filter_string,
+            is_cluster=is_cluster,
+        )
+
+    def fetch_group(self, group_id):
+        """Fetch a group from the API based on it's ID.
+
+        Example::
+
+          >>> try:
```

Comment:
`fetch_metric_descriptor` and `fetch_resource_descriptor` raise `NotFound` if the thing does not exist.

Both kinds of methods can be useful — raising `NotFound` or returning `None` — but it's rarely worth providing both. Raising `NotFound` is better if we believe the calling code is more likely to be written with the assumption that the thing exists. I think that's the case here, since it's hard to imagine why code would be retrieving group 1279483 if there wasn't some particular reason to believe it exists. Raising `NotFound` is better than raising `AttributeError` on a `NoneType` object.